src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectricGoldPan.java

```diff
@@ -69,23 +66,27 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
         for (int slot : getInputSlots()) {
             ItemStack item = menu.getItemInSlot(slot);
 
-            if (SlimefunUtils.isItemSimilar(item, gravel, true, false)) {
+            if (item == null || item.getType() == Material.AIR) {
+                continue;
+            }
+
+            if (goldPan.getInputMaterials().contains(item.getType())) {
                 ItemStack output = goldPan.getRandomOutput();
-                MachineRecipe recipe = new MachineRecipe(3 / getSpeed(), new ItemStack[] { gravel }, new ItemStack[] { output });
+
+                MachineRecipe recipe = new MachineRecipe(3 / getSpeed(), new ItemStack[] { new ItemStack(item.getType()) }, new ItemStack[] { output });
```

Comment:
why the unnecessary cloning
```suggestion
                MachineRecipe recipe = new MachineRecipe(3 / getSpeed(), new ItemStack[] { item }, new ItemStack[] { output });
```